test/pstl_offload/general/selected_different_device.pass.cpp

```diff
@@ -0,0 +1,42 @@
+// -*- C++ -*-
+//===----------------------------------------------------------------------===//
+//
+// Copyright (C) 2023 Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#if __SYCL_PSTL_OFFLOAD__ == 1
+#elif __SYCL_PSTL_OFFLOAD__ == 2
+    #undef __SYCL_PSTL_OFFLOAD__
+    #define __SYCL_PSTL_OFFLOAD__ 3
+#elif __SYCL_PSTL_OFFLOAD__ == 3
+    #undef __SYCL_PSTL_OFFLOAD__
+    #define __SYCL_PSTL_OFFLOAD__ 2
+#else
+#error "PSTL offload is not enabled or the selected value is unsupported"
+#endif
+
+#include <algorithm>
+#include "support/utils.h"
+
+// The test assumes that ONEAPI_DEVICE_SELECTOR is set according to __SYCL_PSTL_OFFLOAD__
+// and check that changed __SYCL_PSTL_OFFLOAD__ leads to an exception exactly from parallel algorithm.
+int main() {
+#if __SYCL_PSTL_OFFLOAD__ == 1
```

Comment:
It seems like it would be more correct to return `TestUtils::done(1)`  here. This call will print "Skipped" and return the skip code.

Reply:
Done.